src/main/java/com/serotonin/bacnet4j/obj/TrendLogObject.java

```diff
@@ -204,8 +205,10 @@ public boolean isLogDisabled() {
         return logDisabled;
     }
 
-    public LogBuffer<LogRecord> getBuffer() {
-        return buffer;
+    public void withBuffer(Consumer<LogBuffer<LogRecord>> bufferConsumer) {

```

Comment:
This is one of the 2 changes that make the tests pass, accessing the buffer in a thread safe manner.